src/ui/zcl_abapgit_gui_chunk_lib.clas.abap

```diff
@@ -153,6 +158,100 @@ CLASS zcl_abapgit_gui_chunk_lib IMPLEMENTATION.
+
+    CONSTANTS: lc_regex_msgid_and_msgno TYPE string VALUE `(.{5})` ##NO_TEXT.
+
+    DATA:
+      lv_error_text      TYPE string,
+      lv_longtext        TYPE string,
+      lv_msgid_and_msgno TYPE string,
+      lv_program_name    TYPE syrepid.
+
+
+    CREATE OBJECT ro_html.
+
+    lv_error_text = ix_error->get_text( ).
+    lv_longtext = ix_error->get_longtext( abap_true ).
+
+    REPLACE ALL OCCURRENCES OF cl_abap_char_utilities=>newline
+            IN lv_longtext
+            WITH '<br>'.
+
+    FIND FIRST OCCURRENCE OF REGEX lc_regex_msgid_and_msgno
+         IN lv_longtext
+         SUBMATCHES lv_msgid_and_msgno.
+
+    IF sy-subrc = 0.
+      REPLACE FIRST OCCURRENCE OF REGEX lc_regex_msgid_and_msgno
+              IN lv_longtext
+              WITH ``.
+    ENDIF.
+
+    REPLACE FIRST OCCURRENCE OF REGEX |(<br>{ zcl_abapgit_message_helper=>gc_section_text-cause }<br>)|
+            IN lv_longtext
+            WITH |<h3>$1</h3>|.
+
+    REPLACE FIRST OCCURRENCE OF REGEX |(<br>{ zcl_abapgit_message_helper=>gc_section_text-system_response }<br>)|
+            IN lv_longtext
+            WITH |<h3>$1</h3>|.
+
+    REPLACE FIRST OCCURRENCE OF REGEX |(<br>{ zcl_abapgit_message_helper=>gc_section_text-what_to_do }<br>)|
+            IN lv_longtext
+            WITH |<h3>$1</h3>|.
+
+    REPLACE FIRST OCCURRENCE OF REGEX |(<br>{ zcl_abapgit_message_helper=>gc_section_text-sys_admin }<br>)|
+            IN lv_longtext
+            WITH |<h3>$1</h3>|.
+
+    ro_html->add( |<div id="message" class="message-panel">| ).
+    ro_html->add( |{ lv_error_text }| ).
+    ro_html->add( |<div class="float-right">| ).
+
+    ro_html->add_a(
+        iv_txt   = `&#x274c;`
+        iv_act   = `toggleDisplay('message')`
+        iv_class = `close-btn`
+        iv_typ   = zif_abapgit_html=>c_action_type-onclick ).
+
+    ro_html->add( |</div>| ).
+
+    ro_html->add( |<div class="float-right message-panel-commands">| ).
+
+    IF lv_msgid_and_msgno IS NOT INITIAL.
+      ro_html->add_a(
+          iv_txt = lv_msgid_and_msgno
+          iv_typ = zif_abapgit_html=>c_action_type-sapevent
+          iv_act = zif_abapgit_definitions=>c_action-goto_message
+          iv_id  = `a_goto_message` ).
+    ENDIF.
+
+    ix_error->get_source_position(
+      IMPORTING
+        program_name = lv_program_name ).
+
+    ro_html->add_a(
+        iv_txt   = `Goto source`
+        iv_act   = zif_abapgit_definitions=>c_action-goto_source
+        iv_typ   = zif_abapgit_html=>c_action_type-sapevent
+        iv_title = |{ lv_program_name }|
+        iv_id    = `a_goto_source` ).
+
+    ro_html->add_a(
+        iv_txt = `Callstack`
+        iv_act = zif_abapgit_definitions=>c_action-show_callstack
+        iv_typ = zif_abapgit_html=>c_action_type-sapevent
+        iv_id  = `a_callstack` ).
+
+    ro_html->add( |</div>| ).
+    ro_html->add( |<div class="message-panel-commands">| ).
+    ro_html->add( |{ lv_longtext }| ).
+    ro_html->add( |</div>| ).
```

Comment:
`if lv_longtext is not initial` ? Also probably style should be named differently.

And ... I'd add a preformatted style for long text instead of inserting `<br>`s above. Maybe... It is not without pitfalls ...

```css
.message-panel-longtext {
  display: none;
  font-family: monospace;
  white-space: pre;
  padding-top: 0.5em;
  font-size: small;
  color: hsl(0, 0%, 40%);
}
div.message-panel:hover .message-panel-longtext {
  display: block
}
```

![image](https://user-images.githubusercontent.com/15635498/61726842-0223e400-ad7b-11e9-8af9-f1854630212f.png)